system/modules/core/drivers/DC_Folder.php

```diff
@@ -2397,8 +2397,18 @@ protected function generateTree($path, $intMargin, $mount=false, $blnProtected=f
 				{
 					$_height = ($objFile->height < 50) ? $objFile->height : 50;
 					$_width = (($objFile->width * $_height / $objFile->height) > 400) ? 90 : '';
+					$imageObj = \Image::create($currentEncoded, array($_width, $_height, 'crop'));
+					$origPath = $imageObj->executeResize()->getResizedPath();
+					$importantPart = $imageObj->getImportantPart();
 
-					$thumbnail .= '<br><img src="' . TL_FILES_URL . \Image::get($currentEncoded, $_width, $_height) . '" alt="" style="margin:0 0 2px -19px">';
+					if ($importantPart['x'] > 0 || $importantPart['y'] > 0)
```

Comment:
Because the important part could be in the top left corner of the image, we should also check for `$importantPart['width']` and `$importantPart['height']`:

```php
if ($importantPart['x'] > 0 || $importantPart['y'] > 0 || $importantPart['width'] < $objFile->width || $importantPart['height'] < $objFile->height)
```